webapp/src/ts/modals/edit-message-group/edit-message-group.component.ts

```diff
@@ -0,0 +1,154 @@
+import { AfterViewChecked, AfterViewInit, Component, OnDestroy, OnInit } from '@angular/core';
+import * as moment from 'moment';
+import { BsModalRef } from 'ngx-bootstrap/modal';
+import { v4 as uuid } from 'uuid';
+
+import { EditGroupService } from '@mm-services/edit-group.service';
+import { SettingsService } from '@mm-services/settings.service';
+import { MmModalAbstract } from '@mm-modals/mm-modal/mm-modal';
+
+@Component({
+  selector: 'edit-message-group',
+  templateUrl: './edit-message-group.component.html',
+})
+export class EditMessageGroupComponent extends MmModalAbstract
+  implements AfterViewInit, OnInit, AfterViewChecked, OnDestroy {
+
+  constructor(
+    bsModalRef:BsModalRef,
+    private editGroupService:EditGroupService,
+    private settingsService:SettingsService,
+  ) {
+    super(bsModalRef);
+  }
+
+  model:any = { group: { rows: [] } };
+
+  private shouldInitDatePickers = false;
+  private settingsPromise;
+  private datePickers = {};
+
+  ngOnInit() {
+    this.getSettings();
+    // task_id variable is only used within this component, to uniquely identify each task in the group
+    this.model?.group?.rows?.forEach(row => row.task_id = uuid());
+  }
+
+  private getSettings() {
+    this.settingsPromise = this.settingsService.get();
```

Comment:
The method is private, single line, and only called once, would be better just to put inline `this.settingsPromise = this.settingsService.get();` and remove the method?

Reply:
Totally, thanks!